### PR TITLE
release-notes: Add v0.1.0 notes

### PR DESCRIPTION
**Pull request by sameo**
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+### Shared filesystem
+
+We added support for the [virtio-fs](https://virtio-fs.gitlab.io/) shared file
+system, allowing for an efficient and reliable way of sharing a filesystem
+between the host and the `cloud-hypervisor` guest.
+
+See our [filesystem sharing](https://github.com/intel/cloud-hypervisor/blob/master/docs/fs.md)
+documentation for more details on how to use virtio-fs with `cloud-hypervisor`.
+
+### Inital direct device assignment support
```
Comment (rbradford) [inline release-notes.md]: typo: *initial*

File: release-notes.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+`--serial tty --console off` from the command line.
+
+### Unit testing
+
+We now run all unit tests from all our crates directly from our CI.
+
+### Integration tests parallelization
+
+The CI cycle run time has been significantly dropped by refactoring our
+integration tests implementation for running all of them in parallel.
```
Comment (rbradford) [inline release-notes.md]: The CI cycle run time has been significantly reduced by refactoring our
￼ integration tests; allowing them to all be run in parallel.

File: release-notes.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# v0.1.0
+
+This release has been tracked through the [0.1.0 project](https://github.com/intel/cloud-hypervisor/projects/1).
+
+Highlights for `cloud-hypervisor` version 0.1.0 include:
+
+### Shared filesystem
```
Comment (rbradford) [inline release-notes.md]: Should mention virtio-pmem!

Comment (rbradford) [reply release-notes.md]: Should mention we now support loading a bzImage.

File: release-notes.md (hunk 2 above)
Comment (sameo) [reply release-notes.md]: Thanks for the suggestion, fixed.

File: release-notes.md (hunk 3 above)
Comment (sameo) [reply release-notes.md]: Right, I was only looking at the 0.1.0 project cards...
I added a section for both features.

File: release-notes.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+`cloud-hypervisor` guests.
+
+### Userspace IOAPIC
+
+`cloud-hypervisor` supports a so-called split IRQ chip implementation by
+implementing support for the [IOAPIC](https://wiki.osdev.org/IOAPIC).
+By moving part of the IRQ chip implementation from kernel space to user space,
+the IRQ chip emulation does not always run in a fully privileged mode.
+
+### Fake persistent memory
```
Comment (rbradford) [inline release-notes.md]: Fake seems the wrong word. Virtual?

File: release-notes.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+The `virtio-pmem` implementation emulates a fake persistent memory device that
+`cloud-hypervisor` can e.g. boot from. Booting from a `virtio-pmem` device
+allows to bypass the guest page cache and improve the guest memory footpring.
+
+### Linux kernel big Zimage (a.k.a. `bzImage`)
```
Comment (rbradford) [inline release-notes.md]: I think you can skip big zImage and just go with bzImage

File: release-notes.md (hunk 4 above)
Comment (sameo) [reply release-notes.md]: Switched to Virtual, thanks.

File: release-notes.md (hunk 5 above)
Comment (sameo) [reply release-notes.md]: Done, thanks.

File: release-notes.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+`cloud-hypervisor` supports a so-called split IRQ chip implementation by
+implementing support for the [IOAPIC](https://wiki.osdev.org/IOAPIC).
+By moving part of the IRQ chip implementation from kernel space to user space,
+the IRQ chip emulation does not always run in a fully privileged mode.
+
+### Virtual persistent memory
+
+The `virtio-pmem` implementation emulates a virtual persistent memory device
+that `cloud-hypervisor` can e.g. boot from. Booting from a `virtio-pmem` device
+allows to bypass the guest page cache and improve the guest memory footpring.
```
Comment (sboeuf) [inline release-notes.md]: s/footpring/footprint/

File: release-notes.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+This release has been tracked through the [0.1.0 project](https://github.com/intel/cloud-hypervisor/projects/1).
+
+Highlights for `cloud-hypervisor` version 0.1.0 include:
+
+### Shared filesystem
+
+We added support for the [virtio-fs](https://virtio-fs.gitlab.io/) shared file
+system, allowing for an efficient and reliable way of sharing a filesystem
+between the host and the `cloud-hypervisor` guest.
```
Comment (sboeuf) [inline release-notes.md]: Should we mention that it is not a complete support of virtio-fs? I mean to say that we don't support the shared memory region which is DAXable and allow for very good performances.

Comment (sameo) [reply release-notes.md]: Maybe we should have that in the `docs/fs.md` file instead?

File: release-notes.md (hunk 6 above)
Comment (sameo) [reply release-notes.md]: Fixed, thanks.

File: release-notes.md (hunk 7 above)
Comment (sboeuf) [reply release-notes.md]: Yes that'd be better indeed.

**Pull request merged**